Source/GeneralErrorView.swift

```diff
@@ -166,8 +167,26 @@ class GeneralErrorView: UIView {
         }
     }
     
-    func setErrorMessage(message: String? = nil) {
+    func setErrorMessage(message: String? = nil, imageName: String? = nil, buttonTitle: String? = nil) {
         errorLabel.attributedText = errorLabelstyle.attributedString(withText: message ?? Strings.Dashboard.generalErrorMessage)
+        
+        if let image = UIImage(named: imageName ?? "") {
+            errorImageView.image = image
+        }
+        
+        if let buttonTitle = buttonTitle {
+            errorActionButton.setAttributedTitle(buttonTitleStyle.attributedString(withText: buttonTitle), for: UIControl.State())
+        }
+    }
+    
+    func showOutdatedVersionError() {
+        errorLabel.attributedText = errorLabelstyle.attributedString(withText: Strings.VersionUpgrade.outDatedMessage)
+        
+        if let image = UIImage(named: "app_update_image") {
+            errorImageView.image = image
+        }
+        
+        errorActionButton.setAttributedTitle(buttonTitleStyle.attributedString(withText: Strings.Coursedates.calendarShiftPromptUpdateNow), for: UIControl.State())
```

Comment:
as the fields are optional, they can be refactored as
```suggestion
        func setErrorMessage(message: String? = nil, imageName: String? = nil, buttonTitle: String? = nil) {
        errorLabel.attributedText = errorLabelstyle.attributedString(withText: message ?? Strings.Dashboard.generalErrorMessage)
        errorImageView.image = UIImage(named: imageName ?? "")
        errorActionButton.setAttributedTitle(buttonTitleStyle.attributedString(withText: buttonTitle), for: UIControl.State())
```

Reply:
```
if let image = UIImage(named: "app_update_image") {
            errorImageView.image = image
        }
```
Deliberately, going with `if let` because in case if the `imageName`  will be nil, the default image will be there. And same is the case with buttonTitle.